### PR TITLE
Fix warnings

### PR DESCRIPTION
**Pull request by malfet**
Add `-Wno-writable-strings`(which is clang's flavor of `-Wwrite-strings`) to list of warnings ignored while compiling torch_python.
Avoid unnecessary copies in range loop
Fix number of signed-unsigned comparisons

Found while building locally on M1

### CODE REVIEW

File: aten/src/ATen/cpu/FlushDenormal.cpp [hunk 1]
```diff
@@ -24,8 +24,12 @@ bool set_flush_denormal(bool on) {
     _mm_setcsr(csr);
     return true;
   }
-#endif
   return false;
 }
+#else
+bool set_flush_denormal(bool on) {
+  return false;
```
Comment (albanD) [inline aten/src/ATen/cpu/FlushDenormal.cpp:31]: Do we want to add a `TORCH_WARN_ONCE()` here as we drop the user flag completely :D

File: torch/csrc/autograd/python_anomaly_mode.h [hunk 2]
```diff
@@ -9,9 +9,9 @@ namespace torch { namespace autograd {
 
 struct PyAnomalyMetadata : public AnomalyMetadata {
   // NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables,clang-diagnostic-writable-strings)
-  static constexpr char* ANOMALY_TRACE_KEY = "traceback_";
+  static constexpr const char* ANOMALY_TRACE_KEY = "traceback_";
```
Comment (albanD) [inline torch/csrc/autograd/python_anomaly_mode.h:12]: Thanks for fixing this!

**Review by albanD — APPROVED** [review]
LGTM

File: aten/src/ATen/cpu/FlushDenormal.cpp (hunk 1 above)
Comment (malfet) [reply aten/src/ATen/cpu/FlushDenormal.cpp:31]: Good point, looks like we simply need to add implementation for aarch64 and throw warning for other arches, such as PowerPC, filed https://github.com/pytorch/pytorch/issues/62983

**Pull request closed without merging**